onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -892,7 +892,8 @@ def _infer_Gather(self, node):
             helper.make_tensor_value_info(node.output[0], self.known_vi_[node.input[0]].type.tensor_type.elem_type,
                                           data_shape[:axis] + indices_shape + data_shape[axis + 1:]))
         # for 1D input, do some sympy compute
-        if node.input[0] in self.sympy_data_ and len(data_shape) == 1 and 0 == get_attribute(node, 'axis', 0):
+        if node.input[0] in self.sympy_data_ and (node.input[1] in self.sympy_data_ or node.input[1] in self.initializers_) \
+           and len(data_shape) == 1 and 0 == get_attribute(node, 'axis', 0):
             idx = self._get_value(node, 1)
```

Comment:
A better fix might be here: change ```_get_value``` to ```_try_get_value```, which returns None if input[1] is unknown

Reply:
Thanks. Will do